app/models/task.py

```diff
@@ -1,5 +1,37 @@
-from sqlalchemy.orm import Mapped, mapped_column
+from sqlalchemy.orm import Mapped, mapped_column, relationship
 from ..db import db
+from sqlalchemy import ForeignKey
+from datetime import datetime
+from typing import TYPE_CHECKING, Optional
+if TYPE_CHECKING:
+    from .goal import Goal
 
 class Task(db.Model):
     id: Mapped[int] = mapped_column(primary_key=True, autoincrement=True)
+    title: Mapped[str]
+    description: Mapped[str]
+    completed_at: Mapped[Optional[datetime]] = mapped_column(nullable=True)
+    goal_id: Mapped[Optional[int]] = mapped_column(ForeignKey("goal.id"))
+    goal: Mapped[Optional["Goal"]] = relationship(back_populates="tasks")
+    
+    def to_dict(self):
+        task = {
+                "id":self.id,
+                "title":self.title,
+                "description":self.description,
+                "is_complete": self.is_complete()
```

Comment:
```suggestion
                "is_complete": True if self.completed_at else False
```

Instead of using a helper function, we could just use a ternary operator here.